### PR TITLE
(docs)(DOC-3458) Note that PE defaults may differ

### PR DESCRIPTION
**Pull request by gguillotte**
Some `puppet.conf` settings are modified in a default PE installation. This leads to some customer confusion, such as assuming the open source configuration reference generated from Puppet code is incorrect (DOCUMENT-803).

This PR adds a bullet point to the top of the configuration reference noting that PE defaults might differ, provides examples, and reminds users to check `puppet.conf` to confirm PE defaults. An equivalent paragraph is also being added to the PE configuration overview.

CC @melamos

### CODE REVIEW

File: lib/puppet/reference/configuration.rb [hunk 1]
```diff
@@ -54,6 +54,13 @@
 
 * Each of these settings can be specified in `puppet.conf` or on the
   command line.
+* Puppet Enterprise (PE) shares configuration settings used in open source
+  Puppet and documented in the configuration reference. However, PE defaults
```
Comment (jhelwig) [inline lib/puppet/reference/configuration.rb]: "`Puppet Enterprise (PE) shares configuration settings used in open source Puppet and documented in the configuration reference.`" feels like there's a verb missing before "`documented`". Should that be "`... used in open source Puppet and is documented in the configuration reference`"? Should "`documented`" be "`documentation`" instead (shares configuration settings and documentation)?

Comment (gguillotte) [reply lib/puppet/reference/configuration.rb]: The two clauses are "used in open source Puppet" and "documented in the configuration reference".

Puppet Enterprise (PE) shares configuration settings / used in open source Puppet

Puppet Enterprise (PE) shares configuration settings / documented in the configuration reference

Comment (kayayarai) [reply lib/puppet/reference/configuration.rb]: Jorie asked me to chime in. How about: "Puppet Enterprise (PE) and open source Puppet share the configuration settings that are documented here. However, PE defaults for some settings differ from the Puppet defaults." etc.

Comment (jhelwig) [reply lib/puppet/reference/configuration.rb]: This had completely fallen off my radar.

@kayayarai 👍 That completely addresses the grammar concerns that I had.

Comment (gguillotte) [reply lib/puppet/reference/configuration.rb]: @kayayarai @jhelwig @jtappa I've revised the statement.

**Comment by jtappa** [comment]
Thanks @gguillotte, if you squash or reword the second commit to have "(docs)" Travis should pass and we can merge

**Comment by gguillotte** [comment]
@jtappa Done, thanks

**Pull request merged**